src/containers/SettingList/Organisation/Organisation.test.tsx

```diff
@@ -48,14 +67,49 @@ test('it renders component in edit mode', async () => {
   // loading is show initially
   expect(getByText('Loading...')).toBeInTheDocument();
 
+  //correct values are rendered in the form
   await waitFor(() => {
-    const phoneNumber = getByTestId('phoneNumber');
+    const inputElements = screen.getAllByRole('textbox');
+    const numberInputElements = screen.getAllByRole('spinbutton');
+
+    const orgName = inputElements[0] as unknown as HTMLInputElement;
+    const signaturePhrase = inputElements[1] as unknown as HTMLInputElement;
+    const phoneNumber = inputElements[2] as unknown as HTMLInputElement;
+    const lowBalanceThreshold = numberInputElements[0] as unknown as HTMLInputElement;
+    const criticalBalanceThreshold = numberInputElements[1] as unknown as HTMLInputElement;
+
     fireEvent.click(phoneNumber);
-    expect(phoneNumber).toBeInTheDocument();
+    expect(orgName?.value).toBe('Glific');
+    expect(signaturePhrase?.value).toBe('Please change me, NOW!');
+    expect(phoneNumber?.value).toBe('917834811114');
+    expect(lowBalanceThreshold?.value).toBe('10');
+    expect(criticalBalanceThreshold?.value).toBe('5');
   });
 
   await waitFor(() => {
     const submit = getByTestId('submitActionButton');
     fireEvent.click(submit);
   });
 });
+
+test('it submits form correctly', async () => {
+  const { getByText } = render(wrapper);
+
+  expect(getByText('Loading...')).toBeInTheDocument();
+
+  await waitFor(() => {
+    const inputElements = screen.getAllByRole('textbox');
+    const numberInputElements = screen.getAllByRole('spinbutton');
+
+    fireEvent.change(inputElements[0], { target: { value: 'Glific' } });
+    fireEvent.change(inputElements[1], { target: { value: 'Please change me, NOW!' } });
+    fireEvent.change(numberInputElements[0], { target: { value: '10' } });
+    fireEvent.change(numberInputElements[1], { target: { value: '5' } });
+  });
+
+  await waitFor(() => {
+    const submitButton = screen.getByText('Save');
+    expect(submitButton).toBeInTheDocument();
+    fireEvent.click(submitButton);
```

Comment:
Events should always be outside of waitFor
and the end of a test should always be an assertion